IDEAS/Buildings/Components/Shading/Interfaces/PartialShadingDevice.mo

```diff
@@ -0,0 +1,93 @@
+within IDEAS.Buildings.Components.Shading.Interfaces;
+partial model PartialShadingDevice
+  extends IDEAS.Buildings.Components.Shading.Interfaces.PartialShading;
+  IDEAS.Buildings.Components.BaseClasses.RadiativeHeatTransfer.ExteriorHeatRadiation skyRadFra(
+    A = A_frame,
+    Tenv_nom = Tenv_nom,
+    epsLw = epsLw_frame,
+    linearise = linRad) if haveBoundaryPorts and haveFrame annotation (
+    Placement(visible = true, transformation(origin = {0, 192}, extent = {{-10, -10}, {10, 10}}, rotation = 0)));
+  IDEAS.Buildings.Components.BaseClasses.ConvectiveHeatTransfer.ExteriorConvection eConFra(
+    A = A_frame,
+    azi = azi,
+    inc = inc,
+    linearise = linCon) if haveBoundaryPorts and haveFrame annotation (
+    Placement(visible = true, transformation(origin = {0, 176}, extent = {{10, -10}, {-10, 10}}, rotation = 0)));
+  IDEAS.Buildings.Components.BaseClasses.RadiativeHeatTransfer.ExteriorSolarAbsorption solAbs(
+  A = A_frame,
+  epsSw = epsSw_frame) if haveBoundaryPorts and haveFrame annotation (
+    Placement(visible = true, transformation(extent = {{10, 150}, {-10, 170}}, rotation = 0)));
+  IDEAS.Buildings.Components.BaseClasses.RadiativeHeatTransfer.ExteriorHeatRadiation skyRad(
+  A = A_glazing,
+  Tenv_nom = Tenv_nom,
+  epsLw = epsLw_glazing,
+  linearise = linRad) if haveBoundaryPorts annotation (
+    Placement(visible = true, transformation(origin = {2, 130}, extent = {{-10, -10}, {10, 10}}, rotation = 0)));
+  IDEAS.Buildings.Components.BaseClasses.ConvectiveHeatTransfer.ExteriorConvection eCon(
+  A = A_glazing,
+  azi = azi,
+   inc = inc,
+   linearise = linCon) if haveBoundaryPorts annotation (
+    Placement(visible = true, transformation(extent = {{12, 104}, {-8, 124}}, rotation = 0)));
+  Modelica.Blocks.Math.Add solDif(k1 = 1, k2 = 1) if haveBoundaryPorts and haveFrame annotation (
+    Placement(visible = true, transformation(origin = {90, 30}, extent = {{-10, -10}, {10, 10}}, rotation = 0)));
+  Modelica.Blocks.Sources.RealExpression TeExpr(y=Te_internal)
+    "Expression for Te"
+    annotation (Placement(transformation(extent={{-68,178},{-48,198}})));
+  Modelica.Blocks.Sources.RealExpression TEnvExpr(y=TEnv_internal)
+    "Expression for TEnv"
+    annotation (Placement(transformation(extent={{-68,160},{-48,180}})));
+protected
+  Modelica.Blocks.Interfaces.RealInput TDryBul_internal = Te_internal
+    "Internal variable to avoid assigning a value to a non-input in the model extends statement";
+  Modelica.Blocks.Interfaces.RealInput TEnv_internal
+    "Internal variable to avoid assigning a value to a non-input in the model extends statement";
+equation
+  connect(TEnv_internal,TEnv);
+  if not haveBoundaryPorts then
+    TEnv_internal = 273.15;
+  end if;
```

Comment:
Shouldn't the connection be part of the if-statement (`if not haveboundaryPorts then` constant `else` connect)? Or is this a more clean implementation?